test/ContractTestContext.ts

```diff
@@ -18,92 +18,121 @@ import type {
-        "CDPosition",
-        "Exchanger",
-        "LeverageEngine",
-        "LeverageAllocator",
-        "PositionToken",
-        "ParameterStore",
-        "VaultOUSD",
-        "LvUSDToken",
-        "Coordinator",
-    ]);
+    [context.owner, context.addr1, context.addr2, context.treasurySigner, context.addr3] = await signers;
+
+    const contractRolesWithDefaults = {
+        defaults: {
+            admin: context.owner.address,
+            executive: context.owner.address,
+            governor: context.owner.address,
+            guardian: context.owner.address,
+        },
+        ...contractRoles,
+    } as ContractRolesWithDefaults;
 
     context.externalOUSD = new ethers.Contract(addressOUSD, abiOUSDToken, context.owner);
     context.externalUSDT = new ethers.Contract(addressUSDT, abiUSDTToken, context.owner);
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore
     context.external3CRV = new ethers.Contract(address3CRV, abi3CRVToken, context.owner);
 
-    [
-        context.cdp,
-        context.exchanger,
-        context.leverageEngine,
-        context.leverageAllocator,
-        context.positionToken,
-        context.parameterStore,
-        context.vault,
-        context.lvUSD,
-        context.coordinator,
-    ] = await Promise.all([
-        contracts.CDPosition.deploy(),
-        contracts.Exchanger.deploy(),
-        contracts.LeverageEngine.deploy(context.owner.address),
-        contracts.LeverageAllocator.deploy(context.owner.address),
-        contracts.PositionToken.deploy(context.owner.address),
-        contracts.ParameterStore.deploy(),
-        contracts.VaultOUSD.deploy(context.externalOUSD.address, "VaultOUSD", "VOUSD"),
-        contracts.LvUSDToken.deploy(),
-        contracts.Coordinator.deploy(),
-    ]) as [
-        CDPosition,
-        Exchanger,
-        LeverageEngine,
-        LeverageAllocator,
-        PositionToken,
-        ParameterStore,
-        VaultOUSD,
-        LvUSDToken,
-        Coordinator
-    ];
+    const contracts = await deployContracts<ArchContracts>({
+        cdp: ["CDPosition"],
+        coordinator: ["Coordinator"],
+        exchanger: ["Exchanger"],
+        leverageAllocator: ["LeverageAllocator"],
+        leverageEngine: ["LeverageEngine"],
+        lvUSD: ["LvUSDToken"],
+        parameterStore: ["ParameterStore"],
+        positionToken: ["PositionToken"],
+        vault: ["VaultOUSD", context.externalOUSD.address, "VaultOUSD", "VOUSD"],
+    }, contractRolesWithDefaults);
+    Object.assign(context, contracts);
+
+    /* temporary list, in the future will just iterate over all contracts: */
+    const contractsWithRoles = [context.positionToken];
+    /* if contracts have derrived role addresses they should exist under their contract name
+       on defaults. defaults should be the expected final roles when deployed to mainnet: */
+    contractRolesWithDefaults.defaults.PositionToken = {
```

Comment:
This means the default PositionToken exec is levEng, which overrides existing default which is owner

Reply:
Is there a way to create object via `ContractTestContext` and not have the levEng as the executive? Other then overriding it again like you do here https://github.com/thisisarchimedes/Archimedes_Finance/pull/78/files#diff-d7504063d17d082a3abc045726c47bfecee67134d4ff6bc7703ce66b666f2b65R21

Reply:
Since ContractTestContext is meant to replicate the deployed state of interactions between contracts I think levEng should be executive here. The defaults object at the top is just meant to create a set of default roles for each contract just so the keys exist to be overridden. I think to make this less confusing I should just move this into that defaults object which will be easier in the next PR. Over time this object will have more default roles corresponding to the contracts that should be executive for each contract.